contracts/CommonBlacklist.sol

```diff
@@ -0,0 +1,375 @@
+     *
+     * @dev Callable by blacklist operator
+     *
+     */
+    function addUsersToBlacklist(
+        address[] memory _users
+    ) external onlyBlacklistOperator {
+        for (uint i; i < _users.length; i++) {
+            require(_users[i] != address(0), "Cannot be zero address");
+
+            blacklist[_users[i]] = true;
+        }
+    }
+
+    /**
+     * @notice Remove users from global blacklist
+     * @param _users: users array for removing from blacklist
+     *
+     * @dev Callable by blacklist operator
+     *
+     */
+    function removeUsersFromBlacklist(
+        address[] memory _users
+    ) external onlyBlacklistOperator {
+        for (uint i; i < _users.length; i++) {
+            blacklist[_users[i]] = false;
+        }
+    }
+
+    /**
+     * @notice Add user to internal blacklist
+     * @param _token: address of token contract
+     * @param _users: users array for adding to blacklist
+     *
+     * @dev Callable by blacklist operator
+     *
+     */
+    function addUsersToInternalBlacklist(
+        address _token,
+        address[] memory _users
+    ) external onlyBlacklistOperator {
+        for (uint i; i < _users.length; i++) {
+            require(_users[i] != address(0), "Cannot be zero address");
+
+            internal_blacklist[_token][_users[i]] = true;
+        }
+    }
+
+    /**
+     * @notice Remove users from internal blacklist
+     * @param _token: address of token contract
+     * @param _users: users array for removing from blacklist
+     *
+     * @dev Callable by blacklist operator
+     *
+     */
+    function removeUsersFromInternalBlacklist(
+        address _token,
+        address[] memory _users
+    ) external onlyBlacklistOperator {
+        for (uint i; i < _users.length; i++) {
+            internal_blacklist[_token][_users[i]] = false;
+        }
+    }
+
+    /**
+     * @notice Function returns current day
+     */
+    function getCurrentDay() public view returns(uint256) {
+        return block.timestamp / 1 days;
+    }
+
+    /**
+     * @notice Function returns current month
+     */
+    function getCurrentMonth() public view returns(uint256) {
+        return block.timestamp / 30 days;
+    }
+
+    /**
+     * @notice Setting token limits
+     * @param _dayLimit: day limit for token transfer
+     * @param _monthLimit: month limit for token transfer
+     *
+     * @dev Callable by blacklist operator
+     *
+     */
+    function settingTokenLimits(
+        address _token,
+        uint256 _dayLimit,
+        uint256 _monthLimit
+    ) external onlyBlacklistOperator {
+        token_limits[_token] = TokenLimit(_dayLimit, _monthLimit);
+        tokens_with_limits[_token] = _dayLimit != 0 || _monthLimit != 0;
+
+        emit SetTokenLimit(_token, _dayLimit, _monthLimit);
+    }
+
+    /**
+     * @notice Save user transfers
+     * @param _user: user address
+     * @param _amount: amount of tokens
+     *
+     */
+    function saveUserTransfers(
```

Comment:
too many calls to this contract, we can ckeck daily and monthly limits here, reducing 2 calls.